pkg/controller/convert.go

```diff
@@ -0,0 +1,173 @@
+package controller
```

Comment:
I'll add some documentation in the synchronize file in the other PR. Since synchronize.go is the API of the package, I feel that is the place it should be